### PR TITLE
paramdigger: Add refactorings for ParamDigger

### PR DESCRIPTION
**Pull request by ArkaprabhaChakraborty**
related to zaproxy/zaproxy#7140
Signed-off-by: ArkaprabhaChakraborty <chakrabortyarkaprabha998@gmail.com>

### CODE REVIEW

File: addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/ExtensionParamDigger.java [hunk 1]
```diff
@@ -109,15 +109,15 @@ public void unload() {
...
         // TODO change the display name based on the config.
         scanController.startScan("Scan", config);
     }
 
-    private ParamMinerPanel getParamMinerPanel() {
+    private ParamDiggerPanel getParamMinerPanel() {
         if (paramMinerPanel == null) {
             paramMinerPanel =
-                    new ParamMinerPanel(scanController, options, () -> showParamMinerDialog(null));
+                    new ParamDiggerPanel(scanController, options, () -> showParamMinerDialog(null));
```
Comment (kingthorin) [inline addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/ExtensionParamDigger.java:120]: showParamMinerDialog

File: addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/ExtensionParamDigger.java [hunk 2]
```diff
@@ -139,7 +139,7 @@ private ZapMenuItem getMenu() {
     public void showParamMinerDialog(HttpMessage node) {
```
Comment (kingthorin) [inline addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/ExtensionParamDigger.java:139]: showParamMinerDialog

**Comment by thc202** [comment]
Is the package going to be renamed later?

File: addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/ExtensionParamDigger.java [hunk 3]
```diff
@@ -25,29 +25,29 @@
...
-    private ParamMinerOptions options;
-    private ParamMinerPanel paramMinerPanel;
+    private ParamDiggerOptions options;
+    private ParamDiggerPanel paramMinerPanel;
     private ExtensionPopupMenuItem paramMinerDialogPopMenu;
     private ZapMenuItem menu;
-    private ParamMinerAPI api;
-    private ParamMinerDialog paramMinerDialog;
+    private ParamDiggerAPI api;
+    private ParamDiggerDialog paramMinerDialog;
```
Comment (thc202) [inline addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/ExtensionParamDigger.java:46]: The variables were not renamed.

**Comment by ArkaprabhaChakraborty** [comment]
> Is the package going to be renamed later?

yes

**Comment by ArkaprabhaChakraborty** [comment]
I think that should be it  :)

**Review by kingthorin — APPROVED** [review]
I haven't pulled the branch to look for others, but what's changed here seems fine to me.

**Comment by thc202** [comment]
The exception message in `ParamGuesserScanController#startScan` needs to be updated.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**